MapboxSceneKit/TerrainNode.swift

```diff
@@ -31,6 +31,7 @@ open class TerrainNode: SCNNode {
     fileprivate static let rgbTileSize = CGSize(width: 256, height: 256)
     fileprivate static let styleTileSize = CGSize(width: 256, height: 256)
 
+    private static let maxTextureImageSizeInBytes: Int = (1024 * 1024) // set a max texture size in order to dynamically calculate the highest zoom level for a given lat/lon bounding rect. Need to balance download speed and detail, so set to 1MB for now
```

Comment:
Maybe it's worth considering to make this one public? Probably it will be different for various devices and people may want to override it. I guess it might be calculated based on the screen size and memory possibilities of the device AND the fact how much memory does usually the client app take. In case it's crashing the dev probably should be able to decrease it, if I get the concept properly?

Reply:
@natalia-osa - If you like the new PR, then I will merge it to master.

As far as line drawing, it still needs work. if what's in #29 , then you could use it in your app, otherwise I would stick to something like the approach you are already using.

We hope to get the line drawing done soon, it's just been delayed due to other work.

Reply:
Well, my approach is fully working, so probably for the release, we'll go with it until you're finished with your solution. I consider your way to be much more performant than mine, but mine obviously was much faster to implement. Once your is ready, mine won't be needed any longer, up to then I don't have an issues with using mine.

Reply:
I'm happy with this PR. After my tests, it's perfect to merge it.

Reply:
Are you ok with my latest revision, or do you prefer to go with the 'maxTextureImageSizeInBytes' version?

Reply:
@avi-c I'm happy with your latest revision.